activerecord/lib/active_record/table_metadata.rb

```diff
@@ -62,6 +62,18 @@ def associated_table(table_name)
       TableMetadata.new(association_klass, arel_table, association)
     end
 
+    def table_to_assoc(table_name)
+      klass._reflect_on_association(table_name) ||
+          klass._reflect_on_association(table_name.to_s.singularize) ||
+          klass.reflect_on_all_associations.detect do |assoc|
+            begin
+              assoc.table_name == table_name
+            rescue NameError
+              false
```

Comment:
is it better to return `nil`?

Reply:
Thx for the comment. It's inside detect, so I'd say boolean here.

Reply:
sorry, you are right

Reply:
no worries :)